src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -236,6 +241,60 @@ public ObservableCollection<string> LanguagesList
             }
         }
 
+        /// <summary>
+        /// PackagePathsForInstall contains the list of all package paths where
+        /// packages can be installed.
+        /// </summary>
+        public ObservableCollection<string> PackagePathsForInstall
+        {
+            get
+            {
+                if (packagePathsForInstall == null || !packagePathsForInstall.Any())
+                {
+                    var programDataPath = new Uri(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData) + "\\");
+                    // Filter Std lib and ProgramData paths from list of paths for download
+                    var customPaths = preferenceSettings.CustomPackageFolders.Where(
+                        x => x != DynamoModel.StandardLibraryToken && !programDataPath.IsBaseOf(new Uri(x)));
+
+                    packagePathsForInstall = new ObservableCollection<string>();
+                    foreach (var path in customPaths)
+                    {
+                        var attr = File.GetAttributes(path);
+
+                        // Add only directory paths
+                        if ((attr & FileAttributes.Directory) == FileAttributes.Directory)
+                            packagePathsForInstall.Add(path);
+                    }
+                }
+                return packagePathsForInstall;
+            }
+            set
+            {
+                packagePathsForInstall = value;
+                RaisePropertyChanged(nameof(PackagePathsForInstall));
+            }
+        }
+
+        /// <summary>
+        /// Currently selected package path where new packages will be downloaded.
+        /// </summary>
+        public string SelectedPackagePathForInstall
+        {
+            get
+            {
+                return selectedPackagePathForInstall;
```

Comment:
couldn't we just use preferenceSettings.selectedPackagePathForInstall instead of the private selectedPackagePathForInstall ?

Reply:
You mean simply to have the public property and remove the private backing field? How would we then implement the getter and setter?

Reply:
I think @pinzart90 means to use the `preferenceSettings.selectedPackagePathForInstall` as the backing field.